src/ECModel.jl

```diff
@@ -852,6 +853,187 @@ function split_financial_terms(ECModel::AbstractEC, profit_distribution=nothing)
+        PEAK=Ann_peak_charges,
+        EN_SELL=Ann_energy_revenues,
+        EN_CONS=Ann_energy_costs,
+        EN_NET=Ann_net_energy_costs,
+    )
+end
+
+function business_plan_dataframe(ECModel::AbstractEC,profit_distribution=nothing, user_set_financial=nothing)
+    gen_data = ECModel.gen_data
+    
+    project_lifetime = field(gen_data, "project_lifetime")
+    year_set = 1:project_lifetime
+    
+    if isnothing(user_set_financial)
+        user_set_financial = [EC_CODE; get_user_set(ECModel)]
+    end
+    if isnothing(profit_distribution)
+        user_set = get_user_set(ECModel)
+        profit_distribution = JuMP.Containers.DenseAxisArray(
+            [ 0.0
+                for y in year_set, u in setdiff(user_set_financial, [EC_CODE])
+            ]
+            , year_set, user_set,
+        )
+    end
+
+    # Create a vector of years from 2023 to (2023 + project_lifetime)
+    gen_data = ECModel.gen_data
+    project_lifetime = field(gen_data, "project_lifetime")
+    years = 1:project_lifetime
+
+    business_plan = split_yearly_financial_terms(ECModel)
+
+    # Create an empty DataFrame
+    df_business = DataFrame(Year = Int[], CAPEX = Float64[], OEM = Float64[], EN_SELL = Float64[], EN_CONS = EN_SELL = Float64[], REP = Float64[], 
+    REWARD = Float64[], RV = Float64[], PEAK = Float64[])
+    for i in years
+        CAPEX = sum(business_plan.CAPEX[i, :])
+        Year = 2022 + years[i]
+        OEM = sum(business_plan.OEM[i, :])
+        EN_SELL = sum(business_plan.EN_SELL[i, :])
+        PEAK = sum(business_plan.PEAK[i, :])
+        REP = sum(business_plan.REP[i, :])
+        RV = sum(business_plan.RV[i, :])
+        REWARD = sum(business_plan.REWARD[i, :])
+        EN_CONS = sum(business_plan.EN_CONS[i, :])
+        push!(df_business, (Year, CAPEX, OEM, PEAK, REP, REWARD, EN_SELL, EN_CONS, RV))
+    end
+
+    return df_business
+end
+
+function business_plan_plot(ECModel::AbstractEC, df_business=nothing)
+    if df_business === nothing
+        df_business = business_plan_dataframe(ECModel)
+    end
+
+    # Extract the required columns from the DataFrame
+    years = df_business.Year
+    capex = -df_business.CAPEX
+    oem = -df_business.OEM
+    en_sell = df_business.EN_SELL
+    en_cons = -df_business.EN_CONS
+    rep = -df_business.REP
+    reward = df_business.REWARD
+    rv = df_business.RV
+    peak = -df_business.PEAK
+
+    # Create a bar plot
+    p = bar(years, [capex, oem, en_sell, en_cons, rep, reward, rv, peak],
+            label=["CAPEX" "OEM" "Energy sell" "Energy consumption" "Replacement" "Reward" "Recovery" "Peak charges"],
+            xlabel="Year", ylabel="Amount [€]",
+            title="Business Over 20 Years",
+            ylims=(0, maximum([capex; oem; en_sell; en_cons; rep; reward; rv; peak])*1.2),
+            legend=:topright,
+            color=:auto,
+            xrotation=45,
+            bar_width=0.6,
+            grid=false,
+            framestyle=:box,
+            )

```

Comment:
For plotting dataframes, there is a direct extension, see https://docs.juliaplots.org/latest/input_data/#DataFrames-support
May be nice to implement in that way

The options of the plot may be moved as optional parameters to the function, so to give easy customizability for different plots

Reply:
I'll have a better look at the repository and I'll try with it :)

Reply:
I would like to add the reward of shared energy which is accounted here because profit distribution is filled with 0. the aggregator should take into consideration not only the sell energy but also the reward
_NPV_agg = ECModel.results[:NPV_agg]  # Annualized profits of the aggregator

Reply:
We can add "using StatsPlots"
 # no need for `using Plots` as that is reexported here as they say in the official Repo: https://github.com/JuliaPlots/StatsPlots.jl